packages/fx-core/src/plugins/resource/frontend/dotnet/plugin.ts

```diff
@@ -60,29 +56,27 @@ export interface DotnetPluginConfig {
 }
 
 export class DotnetPluginImpl implements PluginImpl {
-  config: DotnetPluginConfig = {};
-
-  private syncConfigFromContext(ctx: PluginContext): void {
+  private syncConfigFromContext(ctx: PluginContext): DotnetPluginConfig {
+    const config: DotnetPluginConfig = {};
     const solutionConfig: ReadonlyPluginConfig | undefined = ctx.envInfo.state.get(
       DependentPluginInfo.solutionPluginName
     );
-    this.config.resourceGroupName = solutionConfig?.get(
-      DependentPluginInfo.resourceGroupName
-    ) as string;
-    this.config.subscriptionId = solutionConfig?.get(DependentPluginInfo.subscriptionId) as string;
+    config.resourceGroupName = solutionConfig?.get(DependentPluginInfo.resourceGroupName) as string;
+    config.subscriptionId = solutionConfig?.get(DependentPluginInfo.subscriptionId) as string;
 
-    this.config.webAppName = ctx.config.get(ConfigInfo.webAppName) as string;
-    this.config.appServicePlanName = ctx.config.get(ConfigInfo.appServicePlanName) as string;
-    this.config.projectFilePath = ctx.config.get(ConfigInfo.projectFilePath) as string;
+    config.webAppName = ctx.config.get(ConfigInfo.webAppName) as string;
+    config.appServicePlanName = ctx.config.get(ConfigInfo.appServicePlanName) as string;
+    config.projectFilePath = ctx.projectSettings?.pluginSettings?.projectFilePath as string;
```

Comment:
Nit: 
I'm curious about the change of getting projectFilePath. I guess both the two ways of getting projectFilePath are existing, but the older one is to be deprecated?

Reply:
No, pluginFilePath is irrelevant to environment. It's better to store it in projectSettings rather than envState.